src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/PackagePropertyPage/PackageIconValueProvider.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties.Package
+{
+    [ExportInterceptingPropertyValueProvider(PackageIconPropertyName, ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class PackageIconValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private const string PackageIconPropertyName = "PackageIcon";
+        private const string PackPropertyName = "Pack";
+        private const string PackagePathPropertyName = "PackagePath";
```

Comment:
These fields are "properties" of items. In MSBuild the term "property" has a special meaning, so in order to avoid confusion we refer to these as "metadata".

```suggestion
        private const string PackMetadataName = "Pack";
        private const string PackagePathMetadataName = "PackagePath";
```

Confusingly when we define a XAML rule for an MSBuild item, the metadata there is referred to as properties, but that's due to using rule files for more than one kind of entity. In general, using the term "metadata" for properties of items can avoid confusion.

Reply:
Got it. I'm not sure why I wrote property here. I was thinking the generic xml-terms of `Element` or `Node`, but I used neither of those. My bad. Fixed. (And I updated the PR description accordingly)

Question: Does `Metadata` refer to it, independent if it is an xml `Element` or an `Attribute`? I know that for MSBuild, it generally treats both elements and attributes as the same. But, many xml-based languages differentiate between the two.

Reply:
Yes it's metadata regardless of whether it's expressed as an attribute or child element in the XML representation.